### DISCUSSION TITLE
styling width of a nested div

### DISCUSSION THREAD

**Discussion by DennisWarchocki** [neuefische/questions#253] [Web] [answered]
Hello everyone,

I have troubles with my Navbar. You see the red bar? It's not meant to be red.

<img width="897" alt="Bildschirmfoto 2022-10-12 um 17 01 03" src="https://user-images.githubusercontent.com/109968361/195378245-76aa8c66-885f-45a8-b086-49fbdc8b3b31.png">

I used a Styled Div as a container like this:
	`display: flex;
	width: 100%;
	background-color: red;
	}`

Inside that Div with the red background I nested 3 more Div's with that styling:
`	justify-content: space-evenly;
	padding: 22px 20px;
	background-color: #0bda51;
	color: white;
	font-size: x-large;
	text-decoration: none;
`

I'd like that the 3 children inside the parent div to have all the same space and fill the complete width in green

I tryied many things but couldn't find a solution for me.
Any suggestions?

My repo:
https://github.com/DennisWarchocki/capstone-project

You can find the styled Divs here:  /src/components/styled/ 
one is in StyledNew.js and the other in StyledNav.js

**Comment by DanielGennat** [comment]
Hi Dennis,
- I think `justify-content: space-evenly` has to be in the parent element.
- Change the `background-color` from `red` to `#0bda51` (This could be a quick hack at least)

**Reply by DennisWarchocki** [reply]
Hey Daniel, appreciate your help. You we're right I had to put the space-evenly in the parent element. 
as you mentioned, I could just change the red background to the matching green color inside the children, but I wanted to be able to click everywhere in the navbar.

**Comment by DennisWarchocki** [comment, marked as answer]
The Solution for me was this:

The Parent Div was inside a Styled-component, so I had to set to a <ul> Element here.
`justify-content: space-evenly; padding: 22px 20px; background-color: #0bda51; color: white; font-size: x-large; text-decoration: none;`

I also had to change my Children div which have also been inside a Styled-component to      
 < li> elements.

I also  add here a  `flex-grow: 1` so that the Content inside the Flexbox ( the Navbar Buttons) 
used all Space that was left.

now everything is working fine for me. Thanks to Mareike & Julia in the Office.  =)
